Microsoft.Toolkit.Uwp.UI/AdvancedCollectionView/AdvancedCollectionView.cs

```diff
@@ -464,23 +464,23 @@ private void ItemOnPropertyChanged(object item, PropertyChangedEventArgs e)
                 return;
             }
 
-            var filterResult = _filter(item);
+            var filterResult = _filter?.Invoke(item);
 
-            if (_observedFilterProperties.Contains(e.PropertyName))
+            if (filterResult.HasValue && _observedFilterProperties.Contains(e.PropertyName))
             {
                 var viewIndex = _view.IndexOf(item);
-                if (viewIndex != -1 && !filterResult)
+                if (viewIndex != -1 && !filterResult.Value)
                 {
                     RemoveFromView(viewIndex, item);
                 }
-                else if (viewIndex == -1 && filterResult)
+                else if (viewIndex == -1 && filterResult.Value)
                 {
                     var index = _sourceList.IndexOf(item);
                     HandleItemAdded(index, item);
                 }
             }
 
-            if (filterResult && SortDescriptions.Any(sd => sd.PropertyName == e.PropertyName))
+            if ((filterResult ?? true) && SortDescriptions.Any(sd => sd.PropertyName == e.PropertyName))
```

Comment:
wouldn't `filterResult ?? true` always be true

Reply:
Not quite, as a `bool?`, `filterResult` should have 3 possible values: `null`, `true`, or `false`.

The `??` (null coalescing operator) will only come into play when `filterResult` is null, so we either get `(null ?? true) == true`,  `(true ?? true) == true`, or `(false ?? true) == false`.

The logic here can be broken down like this: 
1.  There is no filter set, so `filterResult` will be `null`, in this case we want to cascade to `true` so that we can check if the property changed matches any of the sort properties.
2. There is a filter set so:
  A. item passes the filter, so the value is `true`, since it will be shown in the view, we want to see if 
the item needs to be resorted.
  B. item fails the filter, so the value is `false`, since it will NOT be shown in the view, we need not evaluate if this item needs to be resorted.

Reply:
Awesome, very helpful :)

Reply:
Thanks for this fix!